src/RepositoryAPI.jl

```diff
@@ -456,9 +137,151 @@ end
 
 
 
+# Internal convenience function.  Updates an object from the repository.
+function postRepositoryObject(token::AbstractString,
+                              objectType::AbstractString,
+                              searchKey::Dict{Symbol, Any};
+                              attributes::Dict=Dict(),
+                              objectFields::Dict=Dict(),
+                              filterRequired::Bool=true
+)
+
+    global verbose
+
+    if token == ""
+        throw(ArgumentError("`token' cannot be empty"))
+    end
+
+    objectID = get(searchKey, :id, 0)
+    name = get(searchKey, :name, "")
+
+    # If objectID is not supplied, retrieve from get function
+    if objectID == 0 
+        if objectType == "alert"
+            object = getRepositoryAlert(token, alertName = name)
+            objectID = get(object, "id", 0)
```

Comment:
since this line is identical across all three, can you just put it outside the if block?